<launch>
  <!-- World setup -->
  <arg name="estimation" default="false"
    doc="set to true to start automatically the computer vision and state estimation nodes." />
  <arg name="debug_sot" default="false"
    doc="set to true to launch the SoT in gdb."/>

  <include file="$(find agimus_demos)/launch/tiago_deburring_world_setup.launch">
    <arg name="gazebo" value="true"/>
  </include>

  <!-- Initialization of robot root_joint pose as a ROS param
       value = "x y z X Y Z W" where Xi+Yj+Zk+W is a quaternion -->
  <!--param name="robot_initial_pose" value="3 0.9 0 0 0 1 0"/-->
  <param name="robot_initial_pose" value="-1 -1 0 0 0 0 1"/>
  <node name="initialize_localization" pkg="agimus_demos" type="tiago_deburring_publish_initial_pose.py"
    output="screen" />

  <!-- SoT parameters -->
  <arg name="script_file" doc="Full path to the script which initialize the supervisor"
    default="$(find agimus_demos)/tiago/deburring/supervisor.py" />

  <include file="$(find agimus_demos)/launch/tiago_deburring_2d_navigation.launch" >
    <arg name="map" value="$(find gerard_bauzil)/maps/201204-gazebo"/>
  </include>

  <include file="$(find tiago_gazebo)/launch/tiago_gazebo.launch" >
    <arg name="world" value="load_plugin_gazebo_ros_link_attacher"/>
    <arg name="robot" value="steel"/>
    <arg name="arm" value="true"/>
    <arg name="end_effector" value="schunk-wsg"/>
    <arg name="ft_sensor"  value="schunk-ft"/>

    <arg name="tuck_arm" value="false"/>
    <arg name="public_sim" value="true"/>
    <!--arg name="gzpose" value="-x 3.0 -y 0.9 -z 0.0 -R 0.0 -P 0.0 -Y 3.14159-->
    <arg name="gzpose" value="-x -1 -y -1 -z 0.0 -R 0.0 -P 0.0 -Y 0
      -J torso_lift_joint 0.15
      -J arm_1_joint 0.1
      -J arm_2_joint -1.47
      -J arm_3_joint -0.16
      -J arm_4_joint 1.87
      -J arm_5_joint -1.57
      -J arm_6_joint 0.01
      -J arm_7_joint 0.0
      -J head_1_joint 0.
      -J head_2_joint 0.
      "/>
    <arg name="extra_gazebo_args" value="--pause"/>
  </include>

  <include file="$(find roscontrol_sot_tiago)/launch/controller.launch" >
    <arg name="use_mobile_base" value="false"/>
    <arg name="end_effector" value="schunk-wsg"/>
    <arg name="simulation" value="true"/>
  </include>

  <!--This starts the HPP interface in namespace agimus/hpp-->
  <group ns="agimus/hpp">
    <node name="hpp_node" pkg="agimus-hpp" type="hpp_node.py"
      args="hpp-manipulation-server" respawn="true"
      output="screen" />
  </group>

  <!--This starts sot supervisor in namespace agimus/sot-->
  <include file="$(find agimus-sot)/launch/supervisor.launch" >
    <arg name="script_file" value="$(arg script_file)"/>
    <arg name="robot_prefix" value="tiago/"/>
    <arg name="simulate_torque_feedback" value="true"/>
    <arg name="required" value="false"/>
  </include>

  <include file="$(find agimus)/launch/path_execution.launch"/>

  <include if="$(arg estimation)" file="$(find agimus_demos)/launch/tiago_deburring_estimation.launch"/>
</launch>
